### PR TITLE
No native git dependency

### PR DESCRIPTION
**Pull request by DavidGOrtega**
Removes native git dep. This effort tries to:
 - avoids future breakage due to git changes
 - avoids having to manage multiple git versions within the code

This effort introduces `init` function in CML. This might be a breaking change for anyone that is using CML as a lib opposed to the common CLI

### CODE REVIEW

**Comment by 0x2b3bfa0** [comment]
> _This might be a breaking change for anyone that is using CML as a lib opposed to the common CLI_

I guess that this breaking change will be the only way to know if that's the case; I doubt it is. 😄

**Comment by 0x2b3bfa0** [comment]
Coining a new concept: fault–induced telemetry.

**Comment by casperdcl** [comment]
> breaking change for anyone that is using CML as a lib

I wouldn't call this a breaking change since "CML as a lib" is not [documented](https://cml.dev/doc) so not (yet) a public API

**Comment by DavidGOrtega** [comment]
I would not  consider it `p2`, I would say `p1` to avoid several disasters like in the past

**Comment by 0x2b3bfa0** [comment]
> _to avoid several disasters like in the past_

...and [potentially] allow for different ones. 😄

File: src/cml.js [hunk 1]
```diff
@@ -69,49 +70,15 @@ const getDriver = (opts) => {
 
   throw new Error(`driver ${driver} unknown!`);
 };
-
-const fixGitSafeDirectory = () => {
```
Comment (0x2b3bfa0) [inline src/cml.js:73]: This also solves the permission issues for any `git` commands ran by users. Should it be ported to the new library?

Comment (DavidGOrtega) [reply src/cml.js:73]: Im testing it. ideally should not be the case. We do not depend on git.

**Comment by DavidGOrtega** [comment]
Im right now stuck on making the build with pkg. This is the [main guilty](https://github.com/vercel/pkg/discussions/1388)
🙁

**Comment by 0x2b3bfa0** [comment]
It's a tough one. Have you considered using Babel to trnaslate all the ESM imports into CJS ones?

**Comment by DavidGOrtega** [comment]
> It's a tough one. Have you considered using Babel to translate all the ESM imports into CJS ones?

I missed this comment.
Yes I tried. I tried to use cjs but its not working at all. Its a kind of a nightmare

**Comment by 0x2b3bfa0** [comment]
As per https://github.com/iterative/cml/issues/1051#issuecomment-1154598734, and lacking https://github.com/vercel/pkg/pull/1323, this pull request will have to stay on CJS unless you find a better workaround. 😅

**Comment by 0x2b3bfa0** [comment]
Note that this pull request should ideally fix https://github.com/iterative/cml/pull/628#discussion_r662371561

**Comment by 0x2b3bfa0** [comment]
@DavidGOrtega & @iterative/cml in general: this pull request is probably going to be blocked for a long time, because https://github.com/iterative/cml/issues/1051#issuecomment-1238754107 is going to take a couple eons to be released. Shall we leave it open until then?

**Comment by 0x2b3bfa0** [comment]
Closing unmerged after leaving a note at https://github.com/iterative/cml/issues/1051#issuecomment-1438829340

**Pull request closed without merging**